jvm/sargon-android/src/main/java/com/radixdlt/sargon/os/storage/EncryptionHelper.kt

```diff
@@ -0,0 +1,146 @@
+@file:OptIn(ExperimentalEncodingApi::class)
+
+package com.radixdlt.sargon.os.storage
+
+import com.radixdlt.sargon.extensions.then
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.security.SecureRandom
+import java.security.spec.AlgorithmParameterSpec
+import javax.crypto.Cipher
+import javax.crypto.SecretKey
+import javax.crypto.spec.GCMParameterSpec
+import javax.crypto.spec.SecretKeySpec
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+
+internal object EncryptionHelper {
```

Comment:
No need for this, at all, right?! We already have AES in Sargon! We should not maintain multiple encryption "implementations".

Reply:
I don't think we can do that. In order to encrypt with Keystore, we need to use Android APIs. Take a closer look at the encrypt method. Indeed the cipher we use is `AES/GCM/NoPadding`, but please take a look at the code:

```
val cipher = Cipher.getInstance(AES_GCM_NOPADDING)
val ivBytes = ByteArray(GCM_IV_LENGTH)
SecureRandom().nextBytes(ivBytes)
val parameterSpec = GCMParameterSpec(AUTH_TAG_LENGTH, ivBytes)
cipher.init(Cipher.ENCRYPT_MODE, secretKey, parameterSpec)

val ciphertext: ByteArray = cipher.doFinal(input)

val byteBuffer = ByteBuffer.allocate(ivBytes.size + ciphertext.size)
byteBuffer.put(ivBytes)
byteBuffer.put(ciphertext)

byteBuffer.array()
```

There is no other way to bypass this API if we want to use Android Keystore.

Reply:
please elaborate...

Reply:
which part is not possible to bypass?

Reply:
The signature is:

```kotlin
 fun encrypt(
        input: ByteArray,
        secretKey: SecretKey
    ): Result<ByteArray> {
```

Which is the same signature as Sargon has:

```rust
    fn encrypt(
        &self,
        plaintext: impl AsRef<[u8]>,
        encryption_key: &mut EncryptionKey,
    ) -> Vec<u8> {
```